contracts/Distributor.sol

```diff
@@ -305,9 +306,20 @@ contract Distributor is UUPSHelper {
     /// @notice Fallback to the last version of the tree
     function _revokeTree() internal {
         MerkleTree memory _tree = lastTree;
-        lastTreeUpdate = 0;
+        endOfDisputePeriod = 0;
         tree = _tree;
-        emit TreeUpdated(_tree.merkleRoot, _tree.ipfsHash);
+        emit Revoked();
+        emit TreeUpdated(
+            _tree.merkleRoot,
+            _tree.ipfsHash,
+            (uint48(block.timestamp) / _EPOCH_DURATION) * (_EPOCH_DURATION) // Last hour
+        );
+    }
+
+    /// @notice Returns the end of the dispute period
+    /// @dev treeUpdate is rounded up to next hour and then `disputePeriod` hours are added
+    function _endOfDisputePeriod(uint48 treeUpdate) internal view returns (uint48) {
+        return ((treeUpdate - 1) / _EPOCH_DURATION + 1 + disputePeriod) * (_EPOCH_DURATION);
```

Comment:
why `treeUpdate - 1` ?

Reply:
to handle the equality case: otherwise 3600 / 3600 + 1 = 2 although we want to round up so we want 1